core/scripts/IntegrationTests.js

```diff
@@ -0,0 +1,567 @@
+        { from: sponsor }
+      );
+      positionsCreated++;
+
+      // STEP 2: transferring tokens to the token holder
+      if (i % 2 == 1) {
+        await syntheticToken.transfer(tokenHolder, baseNumTokens.toString(), {
+          from: sponsor
+        });
+        tokenTransfers++;
+      }
+
+      // STEP 3: advancing time
+      const currentTime = await expiringMultiParty.getCurrentTime();
+      await expiringMultiParty.setCurrentTime(currentTime.add(timeOffsetBetweenTests));
+      await mockOracle.setCurrentTime(currentTime.add(timeOffsetBetweenTests));
+
+      // STEP 4.a: chance to liquidate position. 1 in 3 will get liquidated
+      if (i % 3 == 1 && runLiquidations) {
+        const positionTokensOutstanding = (await expiringMultiParty.positions(sponsor)).tokensOutstanding;
+        await expiringMultiParty.createLiquidation(
+          sponsor,
+          { rawValue: GCR.toString() },
+          { rawValue: positionTokensOutstanding.toString() },
+          { from: liquidator }
+        );
+
+        // get the liquidation info from the event. Used later on to withdraw by accounts.
+        const liquidationEvents = await expiringMultiParty.getPastEvents("LiquidationCreated");
+        const liquidationEvent = liquidationEvents[liquidationEvents.length - 1].args;
+
+        liquidationsObject.push({
+          sponsor: liquidationEvent.sponsor,
+          id: liquidationEvent.liquidationId.toString(),
+          disputed: false
+        });
+
+        // STEP 4.b) Chance to dispute the liquidation. 1 in 2 liquidations will get disputed
+        if (i % 2 == 1 && runDisputes) {
+          // Create the dispute request for the liquidation
+          await expiringMultiParty.dispute(liquidationEvent.liquidationId.toString(), liquidationEvent.sponsor, {
+            from: disputer
+          });
+
+          // Push a price into the oracle. This will enable resolution later on when the disputer
+          // calls `withdrawLiquidation` to extract their winnings.
+          const liquidationTime = await expiringMultiParty.getCurrentTime();
+          await mockOracle.pushPrice(constructorParams.priceFeedIdentifier, liquidationTime, disputePrice);
+
+          liquidationsObject[liquidationsObject.length - 1].disputed = true;
+        }
+      }
+
+      // STEP 5): chance for the token sponsor to deposit more collateral
+      if (i % 2 == 0 && runExtraDeposits) {
+        // Wrap the deposit attempt in a try/catch to deal with a liquidated position reverting deposit
+        try {
+          await expiringMultiParty.deposit({ rawValue: depositAmount.toString() }, { from: sponsor });
+          depositsMade++;
+        } catch (error) {
+          continue;
+        }
+      }
+      // STEP 6): chance for the token sponsor to redeem some collateral
+      if (i % 2 == 1 && runRedeemTokens) {
+        // Wrap the deposit attempt in a try/catch to deal with a liquidated position reverting deposit
```

Comment:
Why not just only do this on iterations when liquidations haven't happened?

Reply:
could do that. they are however discrete processes that dont have anything to do with each other and this set up lets us tune how frequently this will occur.

Reply:
That's a good point. What if you checked the tokens or collateral in the position and skip the redemption if the position is empty? Would that avoid the try catch?

Reply:
Just realized you updated this, so ^ may be moot now.